src/server/database/controllers/messages.js

```diff
@@ -73,42 +67,65 @@ module.exports = {
   getAllToLecture (lecture) {
     return Message.findAll({
       where: {
-        LectureId: lecture.id
-      }
+        LectureId: null
+      },
+      order: '"time" DESC',
+      limit: 10
     })
   },
 
+  
+
+/**
+   * @description Get all 
+   * @returns {Promise.<Message>}
+   */
+  getAll () {
+    return Message.findAll({
+      raw: true,
+      limit: 4,
+      order: '"time" DESC'
+    })
+  },
 
 
+  
   /**
    * @description Retrieve an existing Message by the unique ID
    * @param req
    * @returns {Promise.<Message>}
    */
   retrieve (req) {
     return Message
-      .findById(req.params.MessageId, {
-        include: [{
-          model: Message,
-          as: 'message'
-        }]
+      .find({
+        where: {
+          id: req.id
+        },
       })
   },
 
 
 
   /**
-   * @description
+   * @description Changes voting attributes in message
    * @param req
-   * @returns {Promise.<Message>}
+   * @param callback
+   * @callback Callbacks when voting attributes have been updated
    */
-  vote (req) {
-    var msg = Message.findById(req.id).then(function (result) {
-      if (req.value === -1) {
-        msg.votesDown ++
-      } else {
-        msg.votesUp ++
-      }
-    })
+  vote (req, callback) {
+    return Message.find({where: {id: req.id}})
+      .then(function (msg) {
+        if (req.value === 1) {
+          msg.increment('votesUp', {
+            where: 
+            { id: req.id }
+          })
```

Comment:
Return statamenten vil ikke fungere slik det ser ut som. Må bruk et callback for å vite når vote er ferdig. Kan dermed endre til: 
 ```
msg.increment('votesUp', {
      where: 
             { id: req.id }
       }).then(function{
             if(callback){
                  callback()
             })
       })

```

og tilsvarende på msg.increment under.